File: LoveOnceExt.py
import os,sys,struct

def main():
    if not os.path.exists("data.bin") or not os.path.exists("list.bin"):
        print("File not exist")
        return
    filedata = open("data.bin", 'rb')
    filelist = open("list.bin", 'rb')
    datadata = b'' + filedata.read()
    datalist = b'' + filelist.read()
    pl = 0
    os.system("mkdir data")
    listsize = os.path.getsize("list.bin")
    while pl < listsize:
        filelength,drop,filestart = struct.unpack("<3I",datalist[pl:pl+12])
        readdata = datadata[filestart:filestart+filelength]
        filename = cstr(datalist[pl+16:pl+32])
        wrfile = open("data/"+filename, 'wb')
        wrfile.write(readdata)
        wrfile.close()
        pl += 80
    print("Complete!")

def cstr(s):
    p = "{}s".format(len(s))
    s = struct.unpack(p,s)[0]
    return str(s.replace(b"\x00",b""),encoding = "sjis")
    
if __name__ =="__main__":
    main()
    

File: LoveOnceExt.cpp
#include <iostream>
#include <fstream>

using namespace std;

int main()
{
    ifstream filelist("list.bin", ios::in | ios::binary);
    if (!filelist) {
        std::cout << "List file not found" << endl;
        return -1;
    }
    ifstream filedata("data.bin", ios::in | ios::binary);
    if (!filedata) {
        std::cout << "Data file not found" << endl;
        return -1;
    }
    unsigned int fileinfo[4];
    char filename[16];
    system("mkdir data");
    int filenum = 0;

    while (!filelist.eof()) {
        filelist.clear();
        filelist.seekg(80*filenum);
        if (filelist.eof()) {
            break;
        }

        filelist.read((char*)fileinfo, 16);
        //cout << fileinfo[0] << " " << fileinfo[2] << endl;
        filelist.read((char*)filename, 16);
        //cout << filename << endl;

        char* databyte = (char*)malloc(sizeof(char)*fileinfo[0]);
        filedata.clear();
        filedata.seekg(fileinfo[2]);
        filedata.read(databyte, fileinfo[0]);
        string filepath = filename;
        filepath = "data/" + filepath;
        ofstream filewrite(filepath, ios::out | ios::binary);
        filewrite.write(databyte, fileinfo[0]);
        filewrite.close();
        filenum++;
        free(databyte);
    }


    filelist.close();
    filedata.close();
    std::cout << "Complete!" << endl;
    return 0;
}